packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/Choices.js

```diff
@@ -1,73 +1,123 @@
 import {
-  Box,
-  Grid
+  Box
 } from 'grommet'
 import PropTypes from 'prop-types'
-import React from 'react'
-import sortIntoColumns from 'sort-into-columns'
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
 import howManyColumns from './helpers/howManyColumns'
 import whatSizeThumbnail from './helpers/whatSizeThumbnail'
 import ChoiceButton from './components/ChoiceButton'
 
+const StyledGrid = styled(Box)`
+  display: grid;
+  grid-auto-flow: column;
+  grid-gap: 2px;
+  grid-template-rows: repeat(${props => props.rowsCount}, auto);
+`
+
 function Choices (props) {
   const {
+    autoFocus,
     filteredChoiceIds,
+    handleDelete,
     onChoose,
     selectedChoiceIds,
     task
   } = props
+  const [focusIndex, setFocusIndex] = useState(0)
+
+  useEffect(() => {
+    setFocusIndex(0)
+  }, [filteredChoiceIds])
 
   const columnsCount = howManyColumns(filteredChoiceIds)
-  const sortedFilteredChoiceIds = sortIntoColumns(filteredChoiceIds, columnsCount)
+  const rowsCount = Math.ceil(filteredChoiceIds.length / columnsCount)
   const thumbnailSize = task.alwaysShowThumbnails ? 'small' : whatSizeThumbnail(filteredChoiceIds)
 
+  function handleKeyDown (choiceId, event) {
```

Comment:
I think I can quickly add functionality for Delete (same as Backspace), ArrowRight, and ArrowLeft, though could also leave for subsequent PR.

Reply:
Let's add it, so we have all the keys in one go?

Reply:
I had a solution for ArrowLeft and ArrowRight for most use cases (all within grid, most outside), but a few edge cases (first, last, choices near unfilled grid items) make this more complicated than worth for the time being. 😞